docs/DeveloperGuide.adoc

```diff
@@ -46,177 +39,186 @@ The following class plays an important role at the architecture level:
+and used to run the code in each card.
 
-image::UndoRedoState1.png[]
+To be implemented: An in-app Flashcard Pro compiler and debugger for JS and Java as a playground for the user
 
-Step 3. The user executes `add n/David ...` to add a new person. The `add` command also calls `Model#commitAddressBook()`, causing another modified address book state to be saved into the `addressBookStateList`.
 
-image::UndoRedoState2.png[]
+[[Design-Commons]]
+=== Common classes
 
-[NOTE]
-If a command fails its execution, it will not call `Model#commitAddressBook()`, so the address book state will not be saved into the `addressBookStateList`.
+Classes used by multiple components are in the `dream.fcard.datastructures` package and root of `dream.fcard.util`
 
-Step 4. The user now decides that adding the person was a mistake, and decides to undo that action by executing the `undo` command. The `undo` command will call `Model#undoAddressBook()`, which will shift the `currentStatePointer` once to the left, pointing it to the previous address book state, and restores the address book to that state.
+== Implementation
 
-image::UndoRedoState3.png[]
+This section describes some noteworthy details on how certain features are implemented.
 
-[NOTE]
-If the `currentStatePointer` is at index 0, pointing to the initial address book state, then there are no previous address book states to restore. The `undo` command uses `Model#canUndoAddressBook()` to check if this is the case. If so, it will return an error to the user rather than attempting to perform the undo.
 
-The following sequence diagram shows how the undo operation works:
+// tag::parsejson[]
+==== Implementation
 
-image::UndoSequenceDiagram.png[]
+The parsing is done by `JsonParser`.
+It takes a string and attempts to read it as one of a `JsonValueTypes` and create its corresponding object wrapped in `JsonValue`.
+Since Objects and Arrays can recursively contain a json value, their contents are also read, created and wrapped.
 
-NOTE: The lifeline for `UndoCommand` should end at the destroy marker (X) but due to a limitation of PlantUML, the lifeline reaches the end of diagram.
+* `JsonParser#parseJsonInput()` -- constructs the `JsonValue` object from a string input
 
-The `redo` command does the opposite -- it calls `Model#redoAddressBook()`, which shifts the `currentStatePointer` once to the right, pointing to the previously undone state, and restores the address book to that state.
+Given below is the overview activity diagram of parsing a json string input.
 
-[NOTE]
-If the `currentStatePointer` is at index `addressBookStateList.size() - 1`, pointing to the latest address book state, then there are no undone address book states to restore. The `redo` command uses `Model#canRedoAddressBook()` to check if this is the case. If so, it will return an error to the user rather than attempting to perform the redo.
+image::jsonParserProcessDynamicValue.png[]
```

Comment:
The activity diagram seems to involve some symbol which is not in the CS2103T website.

For example:
![image](https://user-images.githubusercontent.com/39835686/67543919-14b90b00-f726-11e9-8113-c39e29365748.png)

Also, there seems to be missing branch condition like below:
![image](https://user-images.githubusercontent.com/39835686/67543960-35816080-f726-11e9-9d48-dde1d216b155.png)